contracts/extensions/BatchTradeExtension.sol

```diff
@@ -0,0 +1,198 @@
+/*
+    Copyright 2022 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+import { ISetToken } from "@setprotocol/set-protocol-v2/contracts/interfaces/ISetToken.sol";
+import { ITradeModule } from "@setprotocol/set-protocol-v2/contracts/interfaces/ITradeModule.sol";
+
+import { BaseGlobalExtension } from "../lib/BaseGlobalExtension.sol";
+import { IDelegatedManager } from "../interfaces/IDelegatedManager.sol";
+import { IManagerCore } from "../interfaces/IManagerCore.sol";
+
+/**
+ * @title BatchTradeExtension
+ * @author Set Protocol
+ *
+ * Smart contract global extension which provides DelegatedManager privileged operator(s) the ability to execute a
+ * batch of trade on a DEX and the owner the ability to restrict operator(s) permissions with an asset whitelist.
+ */
+contract BatchTradeExtension is BaseGlobalExtension {
+
+    /* ============ Structs ============ */
+
+    struct TradeInfo {
```

Comment:
It might be nice to document these params... I there's something we can copy over from TradeModule here:

https://github.com/SetProtocol/set-protocol-v2/blob/301a5920d722a26c2db229de98f1ee59383cbcd7/contracts/protocol/modules/v1/TradeModule.sol#L110-L114

Reply:
Added!